ethereum_history_api/oracles/src/noir/oracles/transactionOracle/encode.test.ts

```diff
@@ -0,0 +1,50 @@
+import { describe, expect, it } from 'vitest';
+import { TX_OFFSETS, encodeTx } from './encode.js';
+import { loadTxFixture } from '../../../fixtures.js';
+import { ETH_TRANSFER_TX_HASH, CHAIN_LINK_TRANSFER_TX_HASH } from '../../../fixtures/config.js';
+
+// To fix when we have getTransaction for arbitrary length transactions: add test for transaction.to === null.
+// (right now we can't test it because transactions that create contracts (transaction.to === null)
+// have data_len > MAX_TRIE_NODE_LEN)
```

Comment:
`TODO: Add a test for contract creation transaction when we support transactions longer than trie nodes.`